notion-core/src/main/scala/com/ambiata/notion/core/InputOutputLocations.scala

```diff
@@ -0,0 +1,106 @@
+package com.ambiata.notion.core
+
+import com.ambiata.mundane.io.{DirPath, FilePath, FileName}
+import org.apache.hadoop.fs.Path
+
+/**
+ * Type for execution locations.
+ *
+ * An execution can only take place locally or on Hdfs
+ */
+sealed trait ExecutionLocation {
```

Comment:
This type doesn't make a lot of sense to me, how can you have a DirPath a FIlePath and a Path all from a location, some of these must not be safe to call?